reference-architecture/osp-cli/ch4.4_security_groups.sh

```diff
@@ -0,0 +1,129 @@
+done
+
+for PORT in 53 4789 8053 8443 24224 ; do
+    neutron security-group-rule-create master-sg \
+            --protocol udp --port-range-min $PORT --port-range-max $PORT
+done
+
+neutron security-group-rule-create master-sg \
+     --protocol tcp --port-range-min 2380 --port-range-max 2380 \
+     --remote-group-id master-sg
+
+#neutron security-group-rule-create master-sg \
+#     --protocol tcp --port-range-min 8443 --port-range-max 8443 \
+#     --remote-group-id master-sg
+#neutron security-group-rule-create master-sg \
+#     --protocol tcp --port-range-min 8443 --port-range-max 8443 \
+#     --remote-group-id node-sg
+#neutron security-group-rule-create master-sg \
+#     --protocol tcp --port-range-min 8053 --port-range-max 8053 \
+#     --remote-group-id node-sg
+#neutron security-group-rule-create master-sg \
+#     --protocol udp --port-range-min 8053 --port-range-max 8053 \
+#     --remote-group-id node-sg
+
+#
+# etcd:
+#   Port/Proto  From      Reason
+#
+#  2379/TCP     etcd      etcd
+#  2379/TCP     master    etcd
+#  2380/TCP     etcd      etcd
+#neutron security-group-rule-create etcd-sg \
+#     --protocol tcp --port-range-min 2379 --port-range-max 2379 
+#     --remote-group-id etcd-sg
+#neutron security-group-rule-create etcd-sg \
+#     --protocol tcp --port-range-min 2380 --port-range-max 2380 \
+#     --remote-group-id etcd-sg
+
+
+
+# router:
+#   Port/Proto  From      Reason
+#
+#  80/TCP       lb        HTTP
+#  443/TCP      lb        HTTPS
+# infra:
+#   Port/Proto  From      Reason
+#
+#  80/TCP       lb        HTTP
+#  443/TCP      lb        HTTPS
+#neutron security-group-rule-create router-sg \
+#     --protocol tcp --port-range-min 80 --port-range-max 80
+#neutron security-group-rule-create router-sg \
+#     --protocol tcp --port-range-min 443 --port-range-max 443
+
+# infra
+neutron security-group-rule-create infra-sg --protocol icmp
+for PORT in 22 80 443 5000 10250 ; do
+    neutron security-group-rule-create infra-sg \
+            --protocol tcp --port-range-min $PORT --port-range-max $PORT
+done
+neutron security-group-rule-create infra-sg \
+            --protocol tcp --port-range-min 4789 --port-range-max 4789
+
+
+
+# node:
+#   Port/Proto  From      Reason
+#  22/TCP       bastion   Secure Shell
+#  10250/TCP    master    kubelet??
+#  4789/UDP     node      ??
+#neutron security-group-rule-create node-sg --protocol icmp
+#neutron security-group-rule-create node-sg \
+#     --protocol tcp --port-range-min 22 --port-range-max 22 \
+#     --remote-group-id bastion-sg
+#neutron security-group-rule-create node-sg \
+#     --protocol tcp --port-range-min 10250 --port-range-max 10250 \
+#     --remote-group-id master-sg
+#neutron security-group-rule-create node-sg \
+#     --protocol tcp --port-range-min 4789 --port-range-max 4789 \
+#     --remote-group-id node-sg
+neutron security-group-rule-create node-sg --protocol icmp
+for PORT in 22 10250 ; do
```

Comment:
Do we all 22 from 0.0.0.0 or only from the bastion SG? If from 0.0.0.0/0 we should look to restrict to bastion only in the future to gain the full advantages of using a bastion